lib/ansible/module_utils/ec2.py

```diff
@@ -58,6 +60,48 @@ class AnsibleAWSError(Exception):
     pass
 
 
+class JSONFileCache(object):
```

Comment:
There's a code snippet from mixja in https://github.com/boto/botocore/pull/1157#issuecomment-387580482 that looks like a good way to go about this. Ideally this would be enabled for boto3 via config (here are a couple issues tracking that https://github.com/boto/boto3/issues/1179, https://github.com/boto/botocore/issues/1148) and require minimal if any code changes to Ansible. It appears other SDKs keeping parity may be part of the hold-up for that though.